### ISSUE TITLE
Disallow building with unsupported compilers

### ISSUE THREAD

**Issue by Zillode** [syncthing/syncthing#1889]
Should we disallow building with unsupported compilers? It seems we get many issues with segmentation faults, panics etc because xcgo or gccgo is used. For external maintainers that are not reading the tutorial, it could help if we show a warning/error message in this case and provide a `--force` flag to override the error msg.
Good idea?
```
[monitor] 21:20:57 INFO: Syncthing exited: signal: Segmentation fault (core dumped)
[monitor] 21:20:58 INFO: Starting syncthing
[JQ2K7] 21:21:01 INFO: syncthing v0.11.0 (go1.4.2 gccgo (Ubuntu 5.1~rc1-0ubuntu1) 5.0.1 20150414 (prerelease) [gcc-5-branch revision 222102] linux-amd64 default) niklas@Niklas-Netbook 2015-04-26 13:15:08 UTC
[JQ2K7] 21:21:01 INFO: My ID: JQ2K7WZ-JSWDLSH-P56QDWS-CXOG7KP-VU7RZXS-77QDKOC-JXCUG3V-QIT7NAC
[JQ2K7] 21:21:01 INFO: Starting web GUI on http://127.0.0.1:8080/
```

**Comment by calmh** [comment]
The build script does print a warning about it being unsupported and possibly not working...

**Comment by AudriusButkevicius** [comment]
Or do a runtime check and exit immediately

**Comment by calmh** [comment]
```                                                                                 
[monitor] 21:20:57 INFO: Syncthing exited: signal: Segmentation fault (core dumped)
[monitor] 21:20:58 INFO: Starting syncthing
[JQ2K7] 21:21:01 INFO: syncthing v0.11.0 (go1.4.2 gccgo (Ubuntu 5.1~rc1-0ubuntu1) 5.0.1 20150414 (prerelease) [gcc-5-branch revision 222102] linux-amd64 default) niklas@Niklas-Netbook 2015-04-26 13:15:08 UTC
 _    _              _   _                                                       
| |  | |            | | | |  This version of Syncthing has been built with an    
| |__| | ___ _   _  | | | |  outdated, experimental or unsupported compiler. This
|  __  |/ _ \ | | | | | | |  may not work correctly. Before filing a support     
| |  | |  __/ |_| | |_| |_|  request, please consider using an official release  
|_|  |_|\___|\__, | (_) (_)  (see https://syncthing.net/).                       
              __/ |                                                              
             |___/                                                               
[JQ2K7] 21:21:01 INFO: My ID: JQ2K7WZ-JSWDLSH-P56QDWS-CXOG7KP-VU7RZXS-77QDKOC-JXCUG3V-QIT7NAC
[JQ2K7] 21:21:01 INFO: Starting web GUI on http://127.0.0.1:8080/
```

**Comment by AudriusButkevicius** [comment]
:+1: 
```

░░░░▄▄▄▄▀▀▀▀▀▀▀▀▄▄▄▄▄▄
░░░░█░░░░▒▒▒▒▒▒▒▒▒▒▒▒░░▀▀▄
░░░█░░░▒▒▒▒▒▒░░░░░░░░▒▒▒░░█
░░█░░░░░░▄██▀▄▄░░░░░▄▄▄░░░█
░▀▒▄▄▄▒░█▀▀▀▀▄▄█░░░██▄▄█░░░█
█▒█▒▄░▀▄▄▄▀░░░░░░░░█░░░▒▒▒▒▒█
█▒█░█▀▄▄░░░░░█▀░░░░▀▄░░▄▀▀▀▄▒█
░█▀▄░█▄░█▀▄▄░▀░▀▀░▄▄▀░░░░█░░█
░░█░░▀▄▀█▄▄░█▀▀▀▄▄▄▄▀▀█▀██░█
░░░█░░██░░▀█▄▄▄█▄▄█▄████░█
░░░░█░░░▀▀▄░█░░░█░███████░█
░░░░░▀▄░░░▀▀▄▄▄█▄█▄█▄█▄▀░░█
░░░░░░░▀▄▄░▒▒▒▒░░░░░░░░░░█
░░░░░░░░░░▀▀▄▄░▒▒▒▒▒▒▒▒▒▒░█
░░░░░░░░░░░░░░▀▄▄▄▄▄░░░░░█
```

**Comment by Zillode** [comment]
hehe :+1:

**Comment by calmh** [comment]
Not feeling this is a big enough problem that we need to do anything about it at this point.

**Comment by zyxmon** [comment]
@Zillode 
>> Should we disallow building with unsupported compilers? ... For external maintainers that are not reading the tutorial...
Is there any info on using unsupported compilers in wiki? I had to build Syncthing for mipsel(32) my own way. 
I have added the possibility to build go binaries with gccgo to Entware-ng. (now in experimental branch go53 - https://github.com/Entware-ng/Entware-ng/tree/go53 ). Entware-ng is based on openwrt buildroot.
As for building syncthing - the idea is here - https://github.com/Entware-ng/entware-go/blob/master/syncthing/Makefile (still in work). There are no segmentation faults, panics. But there are two main issues.
1. When configuration is changed via web form in many cases syncthing http server stops working and changes are not saved (WARNING: API: use of closed network connection). It does not happen always, it does not happen with STTRACE=all. I think there is some sort of race condition in the code.
2. Disk free space is negative (statfs structure has correct values, but is packed wrong) - this is an issue with gccgo/mipsco32. This checks can be disabled by a patch.
If I will not fix issue (1) I'll probably mark syncthing is broken. I have made a lot of experiments, tried different syncthing versions - there may be no issue in 4 runs, and all other runs have this issue.
I am writing it here just to help others to build syncthing for unsupported compilers.
